modules/daicard/src/components/withdrawSai.js

```diff
@@ -0,0 +1,79 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import { getAddress, formatEther } from "ethers/utils";
+import React, { useState } from "react";
+
+export const WithdrawSaiDialog = ({ channel, machine, saiTokenAddress, saiBalance }) => {
+  const [recipient, setRecipient] = useState("");
+  const [recipientError, setRecipientError] = useState("");
+  const [withdrawing, setWithdrawing] = useState(false);
+
+  const withdrawSai = async () => {
+    let recipientAddress;
+    try {
+      recipientAddress = getAddress(recipient);
+    } catch (e) {
+      setRecipientError("Please enter a valid Ethereum address.");
+      return;
+    }
+    setWithdrawing(true);
+    machine.send("READY");
+    machine.send("START_WITHDRAW");
+    const result = await channel.withdraw({
+      amount: saiBalance.toString(),
+      assetId: saiTokenAddress,
+      recipient: recipientAddress,
+    });
+    console.log(`Cashout result: ${JSON.stringify(result)}`);
+    console.log(`Withdrawing ${saiBalance.toString()} SAI to: ${recipientAddress}`);
+    const txHash = result.transaction.hash;
+    setWithdrawing(false);
+    machine.send("SUCCESS_WITHDRAW", { txHash });
+  };
+
+  return (
+    <Dialog
+      open={machine.state.matches("sai") || withdrawing}
+      aria-labelledby="alert-dialog-title"
+      aria-describedby="alert-dialog-description"
+    >
+      <DialogTitle id="alert-dialog-title">You have SAI in Your Channel!</DialogTitle>
+      <DialogContent dividers>
+        <Typography variant="body1">
+          We've transitioned the DaiCard from SAI (single-collateral DAI) to DAI (multi-collateral
+          DAI)! Please withdraw the SAI from your channel in order to keep using your daicard.
+        </Typography>
+        <Typography variant="h6" component="p">
+          Sai Balance: {saiBalance ? formatEther(saiBalance) : 0}
```

Comment:
we should display this in a way thats more consistent with how we display other balances. 

ie. if you have 2.1 SAI it should display $2.10 (maybe not the symbol, but definitely the decimal place)

Reply:
Will use currency type for this, good call.